ansible_collections/arista/avd/roles/eos_designs/python_modules/base/base.py

```diff
@@ -0,0 +1,664 @@
+        """
+        if (load_interval_default := get(self._hostvars, 'load_interval_default')) is not None:
+            return {'default': load_interval_default}
+        return None
+
+    @cached_property
+    def queue_monitor_length(self):
+        """
+        queue_monitor_length set based on queue_monitor_length data-model and
+        switch.feature_support.queue_monitor_length_notify fact
+        """
+        if (queue_monitor_length := get(self._hostvars, 'queue_monitor_length')) is None:
+            return None
+
+        queue_monitor_length_dict = {'enabled': True}
+        queue_monitor_length_notifying = get(queue_monitor_length, 'notifying')
+        notify_supported = get(self._platform_settings, "feature_support.queue_monitor_length_notify")
+        if (
+                queue_monitor_length_notifying is not None and
+                notify_supported is not False
+        ):
+            queue_monitor_length_dict['notifying'] = queue_monitor_length_notifying
+
+        if get(queue_monitor_length, 'log') is not None:
+            queue_monitor_length_dict['log'] = queue_monitor_length.get('log')
+
+        return queue_monitor_length_dict
+
+    @cached_property
+    def name_server(self):
+        """
+        name_server set based on name_servers data-model and mgmt_interface_vrf
+        """
+        if (name_servers := get(self._hostvars, 'name_servers')) is not None:
+            return {
+                'source':
+                    {
+                        'vrf': self._mgmt_interface_vrf
+                    },
+                'nodes': name_servers
+            }
+
+        return None
+
+    @cached_property
+    def redundancy(self):
+        """
+        redundancy set based on redundancy data-model
+        """
+        if get(self._hostvars, 'redundancy') is not None:
+            return {'protocol': get(self._hostvars, 'redundancy.protocol')}
+        return None
+
+    @cached_property
+    def snmp_server(self):
+        """
+        snmp_server set based on snmp_settings data-model, using various snmp_settings information.
+
+        if snmp_settings.compute_local_engineid is True we will use sha1 to create a
+        unique local_engine_id value based on switch.hostname and switch.mgmt_ip facts.
+
+        If user.version is set to 'v3', compute_local_engineid and compute_v3_user_localized_key are set to 'True'
+        we will use hash_passphrase filter to create an instance of hashlib._hashlib.HASH corresponding to the auth_type
+        value based on various snmp_settings.users information.
+        """
+        if (snmp_settings := get(self._hostvars, 'snmp_settings')) is None:
+            return None
+
+        snmp_server = {}
+        if (compute_local_engineid := snmp_settings.get('compute_local_engineid')) is True:
+            local_engine_id = sha1(f"{self._hostname}{self._mgmt_ip}".encode('utf-8')).hexdigest()
+            snmp_server['engineid'] = {'local': local_engine_id}
```

Comment:
```suggestion
            snmp_server['engine_ids'] = {'local': local_engine_id}
```
@ClausHolbechArista we see that eos_cli_config_gen snmp-settings.j2 expects snmp_server['engine_ids'] as input but eos_designs is outputting snmp_server['engineid']

reference from eos_cli_config_gen/templates/eos/snmp-settings.j2:

```jinja2
{# eos - SNMP Settings#}
{% if snmp_server is arista.avd.defined %}
!
{%     if snmp_server.engine_ids.local is arista.avd.defined %}
snmp-server engineID local {{ snmp_server.engine_ids.local }}
{%     endif %}
```